### PR TITLE
Ignore semver incompatible releases

### LINKED ISSUE #74: Tags that don't follow the Neptune scheme should be ignored
Trying to do a patch release of the [git-hub](https://github.com/sociomantic-tsunami/git-hub/) tool, I got this:

```
object.Exception@./src/semver/Version.d(88): v0.10
----------------
/usr/include/dmd/phobos/std/exception.d:420 pure @safe void std.exception.bailOut!(Exception).bailOut(immutable(char)[], ulong, const(char[])) [0x85b39a]
/usr/include/dmd/phobos/std/exception.d:388 pure @safe bool std.exception.enforce!(Exception, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x85b316]
./src/semver/Version.d:88 semver.Version.Version semver.Version.Version.parse(immutable(char)[]) [0x870fd8]
/home/luca/devel/neptune/src/release/main.d:63 semver.Version.Version release.main.main(immutable(char)[][]).__lambda2!(octod.api.repos.GitRef).__lambda2(octod.api.repos.GitRef) [0x76384c]
/usr/include/dmd/phobos/std/algorithm/iteration.d:597 @property semver.Version.Version std.algorithm.iteration.__T9MapResultS347release4main4mainFAAyaZ9__lambda2TAS5octod3api5repos6GitRefZ.MapResult.front() [0x763a3b]
/usr/include/dmd/phobos/std/array.d:124 semver.Version.Version[] std.array.array!(std.algorithm.iteration.__T9MapResultS347release4main4mainFAAyaZ9__lambda2TAS5octod3api5repos6GitRefZ.MapResult).array(std.algorithm.iteration.__T9MapResultS347release4main4mainFAAyaZ9__lambda2TAS5octod3api5repos6GitRefZ.MapResult) [0x87f4b0]
/home/luca/devel/neptune/src/release/main.d:62 _Dmain [0x760af9]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFNlZv [0x914acb]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x9149fb]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x914a74]
??:? scope void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x9149fb]
??:? _d_run_main [0x91496b]
??:? main [0x76795d]
??:? __libc_start_main [0xc626782f]
```

This tool is from times before neptune and used an incompatible tag `v0.10`. But I don't want to remove this historical tag to be able to use neptune, so I think the tool should ignore tags that don't match the expected format (maybe with a warning? But I'm not sure, as I will get a bogus warning all the time).

PS: Not sure about the v1.0.0 milestone, feel free to change if you don't think is appropriate.

### PR DESCRIPTION
**Pull request by mathias-baumann-sociomantic**
fixes #74

### CODE REVIEW

File: src/release/main.d [hunk 1]
```diff
@@ -60,7 +61,12 @@ void main ( string[] params )
 
     auto con = HTTPConnection.connect(getConf());
     auto repo = con.repository(getUpstream());
-    auto tags = repo.releasedTags().map!(a=>Version.parse(a.name)).array;
+
+    auto tags = repo
+        .releasedTags()
+        .map!(a=>Version.parse(a.name).ifThrown(Version.init))
```
Comment (mihails-strasuns-sociomantic) [inline src/release/main.d]: I'd much rather use `Nullable!Version` (or even algebraic like here https://github.com/sociomantic-tsunami/neptune/blob/master/src/overview/repository.d#L45) to indicate erroneous entries.

**Comment by mathias-baumann-sociomantic** [comment]
Hmm made the code more complicated but as you wish. Changed

File: src/release/main.d [hunk 2]
```diff
@@ -22,6 +22,12 @@ import semver.Version;
 import octod.api.repos;
 import octod.core;
 
+import std.variant;
+
+struct InvalidVersion {}
```
Comment (mihails-strasuns-sociomantic) [inline src/release/main.d]: ddoc ;)

**Comment by mathias-baumann-sociomantic** [comment]
Updated

File: src/release/main.d [hunk 3]
```diff
@@ -22,6 +22,12 @@ import semver.Version;
...
+
+alias VersionAlg = Algebraic!(Version, InvalidVersion);
```
Comment (mihails-strasuns-sociomantic) [inline src/release/main.d]: Please pick more context-meaningful name. Random suggestion `GithubReleaseVersion`.

**Comment by mathias-baumann-sociomantic** [comment]
Renamed

**Pull request merged**